packages/application/src/submission.tsx

```diff
@@ -157,3 +166,62 @@ export class SubmissionHandler {
     );
   }
 }
+
+interface IProps {
+  reason: string;
+  message: string;
+  traceback: string;
+  default_msg: string;
+}
+
+class ErrorDialogContent extends React.Component<IProps, any> {
+  constructor(props: any) {
+    super(props);
+    this.state = { expanded: false };
+  }
+
+  toggleMsgDisplay(): void {
+    // Switch expanded flag
+    const expanded = !this.state.expanded;
+    this.setState({ expanded: expanded });
+  }
+
+  render(): React.ReactElement {
+    const details = this.props.traceback ? (
+      <div>
+        <div>
+          <button
+            className={
+              ERROR_DISPLAY_BUTTON +
+              ' ' +
+              (this.state.expanded ? UP_ICON_CLASS : DOWN_ICON_CLASS)
+            }
+            onClick={(): void => {
+              this.toggleMsgDisplay();
+            }}
+          ></button>
+          {'Error details: '}
+        </div>
+        <br />
+        <div
+          className={
+            this.state.expanded ? ERROR_DETAILS_VISIBLE : ERROR_DETAILS_HIDDEN
+          }
+        >
+          {this.props.traceback}
+        </div>
+      </div>
+    ) : null;
+
+    return (
+      <div className={MESSAGE_DISPLAY}>
+        {this.props.reason + ': ' + this.props.message}
+        <br />
+        <br />
+        {details}
+        <br />
+        <div>{this.props.default_msg}</div>
```

Comment:
Do we still need to include the default message if we're already including the traceback? IIUC we would now be showing everything the user would expect to find in the logs

Reply:
I had that conversation with @kevin-bates and we decided to leave it just because we would very likely still get extra info from the logs, such as prior to the stack trace for instance